synapse/cores/common.py

```diff
@@ -2212,22 +2165,32 @@ def _addDefProps(self, form, fulls):
         for k, v in self.getFormDefs(form):
             fulls.setdefault(k, v)
 
-    def _normTufoProps(self, form, inprops, tufo=None, isadd=False):
+    def _runToAdd(self, fulls):
+        toadd = set()
+
+        for prop, valu in fulls.items():
+            ptype = self.getPropTypeName(prop)
+            for stype in self.getTypeOfs(ptype):
+                if self.isTufoForm(stype):
+                    toadd.add((stype, valu))
+
+        for form, valu in toadd:
+            self.formTufoByProp(form, valu)
+
+    def _normTufoProps(self, form, props, tufo=None, isadd=False):
         '''
         This will both return a set of fully qualified props as a dict
-        as well as modify inprops inband as a normalized set or relatives.
+        as well as modify props inband as a normalized set or relatives.
```

Comment:
"set of relative props"???

Reply:
yea...  mostly me using "set" no in the pythonic sense.  i'll update the doc string.